compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -271,7 +271,7 @@ impl<'a> AstValidator<'a> {
 
         self.session.emit_err(InvalidVisibility {
             span: vis.span,
-            implied: if vis.kind.is_pub() { Some(vis.span) } else { None },
+            implied: vis.kind.is_pub().then(|| vis.span),
```

Comment:
These cases may be simpler as `then_some` (even if longer 🙃 ). I don't have a strong opinion on `then_some` usage, beyond it generating less code